Sources/WalletConnectNetworking/NetworkInteractor.swift

```diff
@@ -36,17 +37,25 @@ public class NetworkingInteractor: NetworkInteracting {
         self.rpcHistory = rpcHistory
         self.logger = logger
         self.socketConnectionStatusPublisher = relayClient.socketConnectionStatusPublisher
-        relayClient.messagePublisher.sink { [unowned self] (topic, message) in
+        setupRelaySubscribtion()
+    }
+
+    private func setupRelaySubscribtion() {
+        relayClient.messagePublisher
+            .asyncFilter { [unowned self] in await topics.contains($0.topic)}
+            .sink { [unowned self] (topic, message) in
             manageSubscription(topic, message)
         }
         .store(in: &publishers)
     }
 
     public func subscribe(topic: String) async throws {
+        await topics.insert(topic)
         try await relayClient.subscribe(topic: topic)
     }
 
     public func unsubscribe(topic: String) {
+        Task { await topics.remove(topic) }
```

Comment:
it could cause some race conditions, mb unsubscribe must be async?

Reply:
priority high if we decide to leave it

Reply:
will add hight priority for now